package/MDAnalysis/analysis/encore/confdistmatrix.py

```diff
@@ -160,73 +149,30 @@ def conformational_distance_matrix(ensemble,
         else:
             subset_masses = None
 
-    # matsize: number of elements of the triangular matrix, diagonal
-    # elements included.
-    matsize = framesn * (framesn + 1) / 2
-
-    # Calculate the number of matrix elements that each core has to
-    # calculate as equally as possible.
-    if ncores > matsize:
-        ncores = matsize
-    runs_per_worker = [matsize / int(ncores) for x in range(ncores)]
-    unfair_work = matsize % ncores
-    for i in range(unfair_work):
-        runs_per_worker[i] += 1
-
-    # Splice the matrix in ncores segments. Calculate the first and the
-    # last (i,j) matrix elements of the slices that will be assigned to
-    # each worker. Each of them will proceed in a column-then-row order
-    # (e.g. 0,0 1,0 1,1 2,0 2,1 2,2 ... )
-    i = 0
-    a = [0, 0]
-    b = [0, 0]
-    tasks_per_worker = []
-    for n,r in enumerate(runs_per_worker):
-        while i * (i - 1) / 2 < np.sum(runs_per_worker[:n + 1]):
-            i += 1
-        b = [i - 2,
-             np.sum(runs_per_worker[0:n + 1]) - (i - 2) * (i - 1) / 2 - 1]
-        tasks_per_worker.append((tuple(a), tuple(b)))
-        if b[0] == b[1]:
-            a[0] = b[0] + 1
-            a[1] = 0
-        else:
-            a[0] = b[0]
-            a[1] = b[1] + 1
-
```

Comment:
joblib takes care now of generating good sized batches to work on. It does some automatic adjustment at the beginning. But we can also give it a rough estimate batchsize to use. This work stealling approach will use all available power until all computations are done. In the old code if a batch was done early the core would just idle until all batches were finished.